src/ai/backend/gateway/logs.py

```diff
@@ -0,0 +1,203 @@
+    log.info('LIST (ak:{0}/{1})',
+             requester_access_key, owner_access_key if owner_access_key != requester_access_key else '*')
+    async with dbpool.acquire() as conn:
+        is_admin = True
+        query = (sa.select('*')
+                   .select_from(error_logs)
+                   .order_by(sa.desc(error_logs.c.created_at))
+                   .limit(params['page_size']))
+        count_query = (sa.select([sa.func.count(error_logs.c.message)])
+                         .select_from(error_logs))
+        if params['page_no'] > 1:
+            query = query.offset((params['page_no'] - 1) * params['page_size'])
+        if request['is_superadmin']:
+            pass
+        elif user_role == UserRole.ADMIN or user_role == 'admin':
+            j = (groups.join(agus, groups.c.id == agus.c.group_id))
+            usr_query = (sa.select([agus.c.user_id])
+                           .select_from(j)
+                           .where([groups.c.domain_name == domain_name]))
+            result = await conn.execute(usr_query)
+            usrs = await result.fetchall()
+            user_ids = [g.id for g in usrs]
+            where = error_logs.c.user_id.in_(user_ids)
+            query = query.where(where)
+            count_query = query.where(where)
+        else:
+            is_admin = False
+            where = ((error_logs.c.user_id == user_uuid) &
+                     (not error_logs.c.is_cleared))
+            query = query.where(where)
+            count_query = query.where(where)
+
+        result = await conn.execute(query)
+        async for row in result:
+            result_item = {
+                'log_id': str(row['id']),
+                'created_at': datetime.timestamp(row['created_at']),
+                'severity': row['severity'],
+                'source': row['source'],
+                'user': row['user'],
+                'is_read': row['is_read'],
+                'message': row['message'],
+                'context_lang': row['context_lang'],
+                'context_env': row['context_env'],
+                'request_url': row['request_url'],
+                'request_status': row['request_status'],
+                'traceback': row['traceback'],
+            }
+            if result_item['user'] is not None:
+                result_item['user'] = str(result_item['user'])
+            if is_admin:
+                result_item['is_cleared'] = row['is_cleared']
+            resp['logs'].append(result_item)
+        resp['count'] = await conn.scalar(count_query)
+        if params['mark_read'].lower() == 'true':
+            update = (sa.update(error_logs)
+                        .values(is_read=True)
+                        .where(error_logs.c.id.in_([x['log_id'] for x in resp['logs']])))
+            await conn.execute(update)
+        return web.json_response(resp, status=200)
+
+
+@auth_required
+@server_status_required(READ_ALLOWED)
+async def clear(request: web.Request) -> web.Response:
```

Comment:
"clear" may be misunderstood as "delete" for the first-time readers.
Let's rename this to "mark_cleared".